src/main/java/org/reaktivity/nukleus/sse/internal/stream/ServerStreamFactory.java

```diff
@@ -814,6 +832,55 @@ private void handleReset(
             final long traceId = reset.trace();
             doReset(applicationReplyThrottle, applicationRouteId, applicationReplyId, traceId);
         }
+
+        private void handleChallenge(
+            ChallengeFW challenge)
+        {
+            final HttpChallengeExFW httpChallengeEx = challenge.extension().get(httpChallengeExRO::tryWrap);
+            if (httpChallengeEx != null)
+            {
+                final JsonObject jsonHeaders = new JsonObject();
+                final JsonObject challengeJson = new JsonObject();
+                final ListFW<HttpHeaderFW> httpHeaders = httpChallengeEx.headers();
+
+                httpHeaders.forEach(header ->
+                {
+                    final String name = header.name().asString();
+                    final String value = header.value().asString();
+                    if (name != null)
+                    {
+                        if (!name.startsWith(":"))
+                        {
+                            jsonHeaders.addProperty(name, value);
+                        }
+                        else if (name.equals(":method"))
+                        {
+                            challengeJson.addProperty("method", value);
+                        }
+                    }
+                });
+                challengeJson.addProperty("headers", gson.toJson(jsonHeaders));
+                writeBuffer.putStringUtf8(0, gson.toJson(challengeJson));
+                final StringFW challengeData = stringRO.wrap(writeBuffer, 0, writeBuffer.capacity());
```

Comment:
You should use a separate `challengeBuffer` for this, since if the `challenge` data is large, it may interfere with `SseEventFW` creation below.